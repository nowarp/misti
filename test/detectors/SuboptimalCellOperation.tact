contract Test {
    fun loadBit() {}

    fun test(s: Slice) {
        s.loadBit(); // Bad
        s.loadInt(4); // Bad

        self.loadBit(); // OK
        let b = s.loadBool(); // OK
    }
}
