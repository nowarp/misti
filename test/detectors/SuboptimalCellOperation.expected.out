[MEDIUM] SuboptimalCellOperation: The result of s.loadBit(); is unused
test/detectors/SuboptimalCellOperation.tact:5:9:
  4 |     fun test(s: Slice) {
> 5 |         s.loadBit(); // Bad
              ^~~~~~~~~~~
  6 |         s.loadInt(4); // Bad
Help: Replace it with `s.skipBit()`
See: https://nowarp.io/tools/misti/docs/detectors/SuboptimalCellOperation

[MEDIUM] SuboptimalCellOperation: The result of s.loadInt(4); is unused
test/detectors/SuboptimalCellOperation.tact:6:9:
  5 |         s.loadBit(); // Bad
> 6 |         s.loadInt(4); // Bad
              ^~~~~~~~~~~~
  7 | 
Help: Replace it with `s.skipInt(4)`
See: https://nowarp.io/tools/misti/docs/detectors/SuboptimalCellOperation